module/models/item-archetype.mjs

```diff
@@ -20,7 +20,10 @@ export default class CrucibleArchetypeItem extends foundry.abstract.TypeDataMode
         obj[ability.id] = new fields.NumberField({...nullableInteger, initial: 2, min: 0, max: 6})
         return obj;
       }, {}), {validate: CrucibleArchetypeItem.#validateAbilities}),
-      talents: new fields.SetField(new fields.DocumentUUIDField({type: "Item"})),
+      talents: new fields.ArrayField(new fields.SchemaField({
+        item: new fields.DocumentUUIDField({type: "Item"}),
+        level: new fields.NumberField({required: true, nullable: true, integer: true, initial: null})
+      })),
```

Comment:
We could consider using a `TypedObjectField` here where the keys are the talent UUIDs and the values are the required levels (or maybe better a schema field if we anticipate other potential metadata). This would have the side effect of avoiding duplicates in the array since the UUID keys would be unique.
```suggestion
      talents: new fields.TypedObjectField(new fields.SchemaField({
        level: new fields.NumberField({required: true, nullable: true, integer: true, initial: null})
      }), {validateKey: itemUuidField.validate}),
```